spec/spec_helper.rb

```diff
@@ -161,6 +161,8 @@ module PuppetSpec
   config.after :each do
     Puppet::Test::TestHelper.after_each_test()
 
+    Puppet.lookup(:environments).clear_all
+
```

Comment:
nit, this should probably go in `Puppet::Test::TestHelper.after_each_test()` as it's something that should be called by puppetlabs_spec_helper when testing modules, and not specific to puppet's spec tests.